CorsixTH/Lua/cheats.lua

```diff
@@ -169,3 +169,50 @@ function Cheats:cheatDecreasePrices()
     end
   end
 end
+
+--! Cheat fax function
+function Cheats:cheatByFax()
+  local cheatWindow = self.hospital.world.ui:getWindow(UICheats)
+  self:announceCheat()
+  -- If a cheats window is open, make sure the UI is updated
+  if cheatWindow then
+    cheatWindow:updateCheatedStatus()
+  end
+end
```

Comment:
I don't quite get the name of the function, how does it cheat by fax? it just shows a cheat became active (or non-active perhaps)

Reply:
Is the new name better?

Reply:
Not really - functions get awkward when you name them based on their context instead of their function, and I'm having a hard time classifying the two actions this function does while also excluding the act of enabling the cheat which is in the parent.

You could *cheat* and call it something like Cheats:_enableCheatCommon; you could write one function for enabling cheats that takes a cheat type and fields out to helper functions for special activation/deactivation logic.

As a loose example (not meant to copy):
```lua
function Cheats:toggleCheat(cheat)
  local hosp = self.hospital
  local ui = hosp.world.ui
  if not hosp.active_cheats[cheat] then
    ui.adviser:say(cheat_messages[cheat].enable)
    hosp.active_cheats[cheat] = true
    self:announceCheat()
    local cheatWindow = hosp.world.ui:getWindow(UICheats)
    if cheatWindow then
     cheatWindow:updateCheatedStatus()
    end
    cheat_fns[cheat].enable(self)
  else
    ui.advistor:say(cheat_messages[cheat].disable)
    cheat_fns[cheat].disable(self)
    hosp.active_cheats[cheat] = nil
  end
end
```